src/backend/distributed/commands/trigger.c

```diff
@@ -517,24 +500,34 @@ ErrorIfUnsupportedDropTriggerCommand(DropStmt *dropTriggerStmt)
 	}
 
 	EnsureCoordinator();
-	ErrorOutForTriggerIfNotCitusLocalTable(relationId);
+	ErrorOutForTriggerIfNotSupported(relationId);
 }
 
 
 /*
- * ErrorOutForTriggerIfNotCitusLocalTable is a helper function to error
+ * ErrorOutForTriggerIfNotSupported is a helper function to error
  * out for unsupported trigger commands depending on the citus table type.
  */
 void
-ErrorOutForTriggerIfNotCitusLocalTable(Oid relationId)
+ErrorOutForTriggerIfNotSupported(Oid relationId)
 {
-	if (IsCitusTableType(relationId, CITUS_LOCAL_TABLE))
+	if (EnableUnsafeTriggers)
```

Comment:
I don't have a strong opinion, but I wonder whether we should still ban triggers from reference tables given how weirdly they behave (called many times).

Reply:
Anyway, we are calling them unsafe, for uniformity, let's keep it for now. As we start using them, we will learn more (and may be discover issues) and we can iterate.